File: app/src/main/java/com/example/androidscript/UITemplate/UIEditor.java
package com.example.androidscript.UITemplate;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import androidx.recyclerview.widget.RecyclerView;

import com.example.androidscript.FloatingWidget.FloatingWidgetService;
import com.example.androidscript.Activities.StartService;
import com.example.androidscript.R;

import java.util.Vector;

public abstract class UIEditor extends Editor {

    protected RecyclerView BlockView;
    protected RecyclerView ButtonView;
    protected Vector<Vector<String>> BlockData;
    protected Vector<String> ButtonData;
    protected String filename;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        this.filename = getIntent().getStringExtra("FileName");
        super.onCreate(savedInstanceState);
        this.setContentView(R.layout.activity_uiactivity);
        this.ButtonView = findViewById(R.id.buttongrid);
        this.BlockView = findViewById(R.id.recycleview);
        this.BlockData = new Vector<>();
        this.ButtonData= new Vector<>();
    }

    protected void startServiceHandler(String orientation){
        if(!FloatingWidgetService.ScriptSet()){
            Toast.makeText(this.getApplicationContext(), "No script loaded!", Toast.LENGTH_LONG).show();
        }
        else if(!StartService.IsAuthorized(this)){
            this.startActivity(new Intent(this, StartService.class).putExtra("Orientation", orientation));
        }else{
            StartService.StartFloatingWidget(this);
        }
    }
}

File: app/src/main/java/com/example/androidscript/uitemplate/UIEditor.kt
package com.example.androidscript.uitemplate

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import androidx.activity.result.ActivityResult
import androidx.activity.result.ActivityResultLauncher
import androidx.activity.result.contract.ActivityResultContracts
import com.example.androidscript.R
import androidx.recyclerview.widget.RecyclerView
import com.example.androidscript.floatingwidget.FloatingWidgetService
import java.util.*

abstract class UIEditor : Editor() {
    protected lateinit var blockView: RecyclerView
    protected lateinit var buttonView: RecyclerView
    protected lateinit var blockData: Vector<Vector<String>>
    protected lateinit var buttonData: Vector<String>
    protected lateinit var fileName: String
    protected abstract val folderName: String
    protected lateinit var resultLauncher: ActivityResultLauncher<Intent>

    override fun onCreate(savedInstanceState: Bundle?) {
        fileName = intent.getStringExtra("FileName")!!
        super.onCreate(savedInstanceState)
        this.setContentView(R.layout.activity_uiactivity)
        buttonView = findViewById(R.id.buttongrid)
        blockView = findViewById(R.id.recycleview)
        blockData = Vector()
        buttonData = Vector()
        resultLauncher =
            registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result: ActivityResult ->
                if (result.resultCode == Activity.RESULT_OK) {
                    this.startService(
                        Intent(this, FloatingWidgetService::class.java)
                            .putExtra(FloatingWidgetService.folderTAG, this.folderName)
                            .putExtra(FloatingWidgetService.scriptTAG, "Run.txt")
                    )
                    this.finishAffinity()
                }
            }
    }
}